File: banchus-packet/src/main/java/pe/nanamochi/banchus/packets/server/ProtocolNegotiationPacket.java
package pe.nanamochi.banchus.packets.server;

import java.io.IOException;
import java.io.OutputStream;
import lombok.Data;
import pe.nanamochi.banchus.core.Packets;
import pe.nanamochi.banchus.core.ServerPacket;
import pe.nanamochi.banchus.io.IDataWriter;

@Data
public class ProtocolNegotiationPacket implements ServerPacket {
  private int protocolVersion;

  public ProtocolNegotiationPacket() {
    this.protocolVersion = 19;
  }

  @Override
  public Packets getPacketType() {
    return Packets.BANCHO_PROTOCOL_NEOGITIATION;
  }

  @Override
  public void write(IDataWriter writer, OutputStream stream) throws IOException {
    writer.writeInt32(stream, protocolVersion);
  }
}

File: banchus-packet/src/main/kotlin/pe/nanamochi/banchus/packets/server/ProtocolNegotiationPacket.kt
package pe.nanamochi.banchus.packets.server

import java.io.OutputStream
import pe.nanamochi.banchus.core.BanchoPacket
import pe.nanamochi.banchus.core.PacketType
import pe.nanamochi.banchus.io.DataWriter

class ProtocolNegotiationPacket(var protocolVersion: Int = 19) :
    BanchoPacket(), BanchoPacket.Server {
    override val type = PacketType.BANCHO_PROTOCOL_NEGOTIATION

    override fun write(writer: DataWriter, out: OutputStream) {
        writer.writeInt32(out, protocolVersion)
    }
}
